### PR TITLE
Backport fix for setuptools 45.0.0+ compatibility & enable macOS builds

### LINKED ISSUE #32: OSX builds failing
https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=26291

```
ld: library not found for -lcrt1.10.6.o
clang-4.0: error: linker command failed with exit code 1 (use -v to see invocation)

Waf: Leaving directory `$SRC_DIR/bootloader/build/debug'
Build failed
 -> task in 'run_d' failed with exit status 1 (run with -v to display more information)
Traceback (most recent call last):
  File "/usr/local/miniconda/bin/conda-build", line 11, in <module>
    sys.exit(main())
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 456, in main
    execute(sys.argv[1:])
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/cli/main_build.py", line 447, in execute
    verify=args.verify, variants=args.variants)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/api.py", line 208, in build
    notest=notest, need_source_download=need_source_download, variants=variants)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/build.py", line 2314, in build_tree
    notest=notest,
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/build.py", line 1477, in build
    cwd=src_dir, stats=build_stats)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/utils.py", line 374, in check_call_env
    return _func_defaulting_env_to_os_environ('call', *popenargs, **kwargs)
  File "/usr/local/miniconda/lib/python3.7/site-packages/conda_build/utils.py", line 354, in _func_defaulting_env_to_os_environ
    raise subprocess.CalledProcessError(proc.returncode, _args)
subprocess.CalledProcessError: Command '['/bin/bash', '-e', '/usr/local/miniconda/conda-bld/pyinstaller_1554337787672/work/conda_build.sh']' returned non-zero exit status 1.
```

Skipping OSX for now so we can get Python 3.7 packages for Windows and Linux, but PRs are welcome.

### PR DESCRIPTION
**Pull request by nehaljwani**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes #32

### CODE REVIEW

**Comment by nehaljwani** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -26,6 +26,8 @@ build:
   script:
     - export LDFLAGS="${LDFLAGS} -L${PREFIX}/lib"               # [unix]
     - pushd bootloader
+    - sed -E -i.bak "s,(os.environ\['MACOSX_DEPLOYMENT_TARGET'\] = )'[0-9]+.[0-9]+',\1'${MACOSX_DEPLOYMENT_TARGET}',g" wscript  # [osx]
+    - sed -E -i.bak "s,(-mmacosx-version-min)=[0-9]+.[0-9]+,\1=${macos_min_version},g" wscript                                  # [osx]
```
Comment (jakirkham) [inline recipe/meta.yaml]: Could you please explain why this `sed` command is needed? Would it be possible to make this a patch instead?

Comment (nehaljwani) [reply recipe/meta.yaml]: I have updated the pull request with a patch for this.

The reason why it is required is that the waf configuration file had hardcoded 10.7 as the minimum macOS version and deployment target, which is way too old to be supported and required crt1.10.6.o which is not present in the 10.9 SDK.

**Comment by nehaljwani** [comment]
@conda-forge/pyinstaller This PR is good to go!

**Comment by nicoddemus** [comment]
Hi @nehaljwani,

First of all thanks for the patch! A question, there are two new patches in your PR. `002` (from you) and `003` which I suppose is from upstream. Should `002` also be applied upstream?

**Comment by nehaljwani** [comment]
I intend to send `002` upstream, wanted it to be reviewed here first.

**Comment by nicoddemus** [comment]
> I intend to send 002 upstream, wanted it to be reviewed here first.

I see, thanks. I myself don't know about this issue specifically, so I will let others review it, but it is definitely a good sign that all builds now pass. 👍

**Comment by nehaljwani** [comment]
@nicoddemus My patch was merged upstream: https://github.com/pyinstaller/pyinstaller/pull/4677

This PR should be good to go now!

**Comment by nehaljwani** [comment]
@conda-forge-admin, please rerender

**Comment by nehaljwani** [comment]
Gentle reminder, @conda-forge/pyinstaller !

**Comment by nicoddemus** [comment]
Thanks @nehaljwani!

**Comment by nehaljwani** [comment]
@nicoddemus Could you hit the merge button too? :stuck_out_tongue_winking_eye:

**Comment by nicoddemus** [comment]
I was waiting for the other maintainers to take a second look. I will approve tomorrow if nobody manifests until then. 👍

**Comment by nicoddemus** [comment]
Thanks @nehaljwani!

**Pull request merged**